log/std/logger_test.go

```diff
@@ -0,0 +1,199 @@
+package std
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+
+	"github.com/beatlabs/patron/log"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewLogger(t *testing.T) {
+	var b bytes.Buffer
+	logger := New(&b, log.InfoLevel, map[string]interface{}{"name": "john doe", "age": 18})
+	assert.NotNil(t, logger.debug)
+	assert.NotNil(t, logger.info)
+	assert.NotNil(t, logger.warn)
+	assert.NotNil(t, logger.error)
+	assert.NotNil(t, logger.fatal)
+	assert.NotNil(t, logger.panic)
+	assert.Equal(t, log.InfoLevel, logger.Level())
+	assert.Equal(t, logger.fields, map[string]interface{}{"name": "john doe", "age": 18})
+	assert.Contains(t, logger.fieldsLine, "age=18")
+	assert.Contains(t, logger.fieldsLine, "name=john doe")
+}
+
+func TestNewSub(t *testing.T) {
+	var b bytes.Buffer
+	logger := New(&b, log.InfoLevel, map[string]interface{}{"name": "john doe"})
+	assert.NotNil(t, logger)
+	subLogger := logger.Sub(map[string]interface{}{"age": 18}).(*Logger)
+	assert.NotNil(t, subLogger.debug)
+	assert.NotNil(t, subLogger.info)
+	assert.NotNil(t, subLogger.warn)
+	assert.NotNil(t, subLogger.error)
+	assert.NotNil(t, subLogger.fatal)
+	assert.NotNil(t, subLogger.panic)
+	assert.Equal(t, log.InfoLevel, subLogger.Level())
+	assert.Equal(t, subLogger.fields, map[string]interface{}{"name": "john doe", "age": 18})
+	assert.Contains(t, subLogger.fieldsLine, "age=18")
+	assert.Contains(t, subLogger.fieldsLine, "name=john doe")
+}
+
+func TestLogger(t *testing.T) {
+	// BEWARE: Since we are testing the log output change in line number of statements affect the test outcome
+	var b bytes.Buffer
+	logger := New(&b, log.DebugLevel, map[string]interface{}{"name": "john doe", "age": 18})
+
+	type args struct {
+		lvl        log.Level
+		msg        string
+		args       []interface{}
+		lineNumber int
+	}
+	tests := map[string]struct {
+		args args
+	}{
+		"debug":  {args: args{lvl: log.DebugLevel, args: []interface{}{"hello world"}, lineNumber: 76}},
```

Comment:
We cannot test with Fatal here because it will `os.Exit(1)`, right?

Reply:
correct